internal/directives/argocd_revisions.go

```diff
@@ -112,6 +112,7 @@ func (a *argocdUpdater) getDesiredRevisionForSource(
 			stepCtx.Freight.References(),
 			repoURL,
 			chartName,
+			false, // mustFind = false means we'll deal with a nil result ourselves
```

Comment:
This is a case where we don't care if we don't find the artifact. It's completely plausible that a source references a chart repo or git repo that our Warehouse(s) don't subscribe to.